src/go/rpk/pkg/config/manager.go

```diff
@@ -116,6 +117,15 @@ func (m *manager) ReadOrGenerate(path string) (*Config, error) {
 		path,
 	)
 	m.v.Set("config_file", path)
+	dir := filepath.Dir(path)
+	err = m.fs.MkdirAll(dir, 0755)
```

Comment:
@alenkacz do you mean testing this? If so, sadly `afero`'s `MemMapFs` doesn't complain if you would write to a dir that doesn't exist, so unit tests don't catch this :/ I'll add a test to ducktape after https://github.com/vectorizedio/redpanda/pull/985 is merged.